visualization/utils/strategies/whole_dataset_mode.py

```diff
@@ -0,0 +1,138 @@
+        super().__init__()
+
+    def get_utils_input_layout(self) -> List[dbc.AccordionItem]:
+        inference_condition = lambda name, value: isinstance(value, (str, int, float))
+        return super().get_utils_input_layout(
+            inference_condition,
+            lambda name, value: True,
+            False,
+            list(self.config.items()),
+        )
+
+    def get_query_input_layout(self) -> List[dbc.AccordionItem]:
+        return []
+
+    def run(self, utils: Dict, params: Dict):
+        runs_storage = get_available_models()
+
+        run_index = len(runs_storage)
+        metrics_directory = RESULTS_PATH.format(run_index)
+        output_file = os.path.join(metrics_directory, OUTPUT_PATH.format(OUTPUT, GREEDY))
+        save_metrics_file = os.path.join(metrics_directory, OUTPUT_PATH.format(METRICS, GREEDY))
+        random_seed_start = utils['start_random_seed'] if params['range_random_mode'] else utils['random_seed']
+        random_seed_end = utils['random_seed'] if params['range_random_mode'] else utils['random_seed'] + 1
+        generate_solutions_config = GenerateSolutionsConfig(
+            output_file=output_file,
+            sandbox=self.config['sandbox'],
+            server=self.config['server'],
+            data_file=self.config['data_file'],
+            **{key: value for key, value in utils.items() if key in current_app.config['data_explorer']},
+        )
+        generate_solutions_config.prompt = PromptConfig(
+            **{key: value for key, value in utils.items() if key in current_app.config['data_explorer']['prompt']}
+        )
+        generate_solutions_config.prompt.context = utils['context_templates']
+        generate_solutions_config.prompt.examples = examples.get(
+            utils['examples_type'] if utils['examples_type'] else "", []
+        )
+
+        generate_solutions_config.inference = InferenceConfig(
+            **{key: value for key, value in utils.items() if key in current_app.config['data_explorer']['inference']}
+        )
+        for random_seed in range(random_seed_start, random_seed_end):
+            output_file = (
+                output_file
+                if not params['range_random_mode']
+                else os.path.join(
+                    metrics_directory,
+                    OUTPUT_PATH.format(OUTPUT, "rs" + str(random_seed)),
+                )
+            )
+            save_metrics_file = (
+                save_metrics_file
+                if not params['range_random_mode']
+                else os.path.join(
+                    metrics_directory,
+                    OUTPUT_PATH.format(METRICS, "rs" + str(random_seed)),
+                )
+            )
+            generate_solutions_config.inference.random_seed = random_seed
+            logging.info("Generate solutions")
+            generate_solutions(OmegaConf.structured(generate_solutions_config))
+
+            evaluate_results_command = evaluate_results_template.format(
```

Comment:
Can you do the same for evaluation? Call a function instead of running script?

Reply:
There is a problem with typings. prediction_jsonl_files should be a str when I do OmegaConf.structured, but in __post_init__ it becomes List[str] and generate_solutions expect it to be List[str]

Reply:
can you set it to Any then if it fails on the type? Or maybe there is some parameter in OmegaConf that you can use to ignore types?